### PR TITLE
[GSoC][TMVA][SOFIE] Add BatchNormalization operator

### PR DESCRIPTION
**Pull request by AaradhyaSaxena**
# This Pull request:
Development of the BatchNormalization operator for the Google Summer of Code 2021 project [Inference Code Generation for Batch Normalization](https://hepsoftwarefoundation.org/gsoc/2021/proposal_TMVAOperatorBatchnorm.html).

## Checklist:

- [ x] tested changes locally

### CODE REVIEW

**Comment by lmoneta** [comment]
@phsft-bot build just on ROOT-ubuntu2004/default with flags -Dtmva-sofie=On

**Review by lmoneta — CHANGES_REQUESTED** [review]
Hi @AaradhyaSaxena 
Thank you for this useful contribution!
Can you please add also a test program for the Batch Normalisation in tmva/sofie/test  ?

**Comment by AaradhyaSaxena** [comment]
> Hi @AaradhyaSaxena
> Thank you for this useful contribution!
> Can you please add also a test program for the Batch Normalisation in tmva/sofie/test ?

@lmoneta, I will add the test program asap.

**Comment by lmoneta** [comment]
This PR is replaced by #9307 , that includes the commits of this PR and extends it adding the test program
